src/protocol.js

```diff
@@ -63,29 +63,36 @@ const authoriseApp = () => {
       joinSchemes: [safeScheme]
     };
     return safeApp.initializeApp(appInfo, netStateChange, opts)
-      .then((app) => app.auth.genConnUri()
+      .then((app) => {
+          appObj = app;
+          return resolve();
+      })
+      .catch(reject);
+  });
+};
+
+const connectSafeApp = () => {
+  return new Promise((resolve, reject) => {
+    if (!appObj) {
+      return reject(new Error('Unexpected error. SAFE App library not initialised'));
+    } else if (appObj.networkState === 'Init') {
```

Comment:
Can we have `Init` defined as a constant?

Reply:
Totally agreed, but we should actually have constants defined in safe_app_nodejs API and use it from here. I created a task for this: https://maidsafe.atlassian.net/browse/MAID-2410